pylint/extensions/private_import.py

```diff
@@ -0,0 +1,190 @@
+            "leading underscores should be considered private.",
+        ),
+    }
+
+    def __init__(self, linter: "PyLinter") -> None:
+        BaseChecker.__init__(self, linter)
+        self.all_used_type_annotations: Set[str] = set()
+        self.populated_annotations = False
+
+    @utils.check_messages("import-private-name")
+    def visit_import(self, node: nodes.Import) -> None:
+        if utils.is_node_in_typing_guarded_import_block(node):
+            return
+        names = [name[0] for name in node.names]
+        private_names = self._get_private_imports(names)
+        if private_names:
+            imported_identifier = "modules" if len(private_names) > 1 else "module"
+            private_name_string = ", ".join(private_names)
+            self.add_message(
+                "import-private-name",
+                node=node,
+                args=(imported_identifier, private_name_string),
+                confidence=HIGH,
+            )
+
+    @utils.check_messages("import-private-name")
+    def visit_importfrom(self, node: nodes.ImportFrom) -> None:
+        if utils.is_node_in_typing_guarded_import_block(node):
+            return
+        private_module_imports = self._get_private_imports([node.modname])
+        if private_module_imports:  # If module is private, do not check imported names
+            self.add_message(
+                "import-private-name",
+                node=node,
+                args=("module", private_module_imports[0]),
+                confidence=HIGH,
+            )
+        elif not self.same_root_dir(
+            node, node.modname
+        ):  # Only check imported names if the module is external
+
+            names = [n[0] for n in node.names]
+            private_names = self._get_private_imports(names)
+            if (
+                not private_names
+            ):  # No private names, don't have to check against typing annotations
+                return
+
+            # Only check names not used as type annotations
+            if not self.populated_annotations:
+                self._populate_type_annotations(
+                    node.root(), self.all_used_type_annotations
+                )
+                self.populated_annotations = True
+
+            private_names = [
+                n for n in private_names if n not in self.all_used_type_annotations
+            ]
+
+            if private_names:
+                imported_identifier = "objects" if len(private_names) > 1 else "object"
+                private_name_string = ", ".join(private_names)
+                self.add_message(
+                    "import-private-name",
+                    node=node,
+                    args=(imported_identifier, private_name_string),
+                    confidence=HIGH,
+                )
+
+    def _get_private_imports(
+        self,
+        names: List[str],
+    ) -> List[str]:
+        """Returns the private names from input names by a simple string check"""
```

Comment:
```suggestion
        """Returns the private names from input names by a simple string check."""
```

I know that `pydocstringformatter` is going to enforce periods at the end a docstring in an upcoming release.